docs/features/populate.rst

```diff
@@ -0,0 +1,296 @@
+Populate
+========
+
+This section explains Robottelo data populate.
+
+.. contents::
+
+Commands
+--------
+Using :code:`$ manage` in the robottelo environment and root directory
+In the subgroup `data` you can find the ``populate`` and ``validate`` commands.
+That commands are used to read data description from YAML file and
+populate the system or validate populated entities.
+
+Read more in specific documentation in :doc:`populate </api/robottelo.populate>`
+
+Having ``test_data.yaml`` with the following content.
+
+  "By default ``tests/foreman/data`` is the path to look for yaml files
+  but you can include a leading :code:`/` to the path ``/tmp/file.yaml``"
+
+.. code-block:: console
+
+    vars:
+      org_label_suffix = inc
+    actions:
+      - model: Organization
+        log: The first organization...
+        register: org_1
+        data:
+          name: MyOrg
+          label: MyOrg{{org_label_suffix}}
+
+
+To populate the system
+
+.. code-block:: console
+
+    (robottelo_env)[you@host robottelo]$ manage data populate test_data.yaml -v
+    2017-01-04 04:31:17 - robottelo.populate.base - INFO - CREATE: The first organization...
+    2017-01-04 04:31:19 - robottelo.populate.base - INFO - search: Organization {'query': {'search': 'name=MyOrg,label=MyOrg'}} found unique item
+    2017-01-04 04:31:19 - robottelo.populate.base - INFO - create: Entity already exists: Organization 36
+    2017-01-04 04:31:19 - robottelo.populate.base - INFO - registry: org_1 registered
+
+To validate the system
+
+.. code-block:: console
+
+    (robottelo_env)[you@host robottelo]$ manage data validate test_data.yaml
+    (robottelo_env)[you@host robottelo]$ echo $?
+    0  # system validated else 1
+
+
+Use :code:`$ manage data --help` and :code:`$ manage populate --help` for more info
+
+Decorator
+---------
+
+Other way to use populate is via decorator, it is useful to decorate a test_case
+forcing a populate or validate operation to be performed.
+
+    Having a data_file like::
+
+        actions:
+          - model: Organization
+            register: organization_1
+            data:
+              name: My Org
+
+    Then you can use in decorators::
+
+        @populate_with('file.yaml')
```

Comment:
sure! you can decorate setUp

```
@populate_with(data, context=True)
def setUp(self, context):
    self.context = context

def test_something(self):
    assert self.context.organization_1.name == "Bla"
```

It can be done and I will add an example to the docs.

(although  it is not recommended, because each test_case should have its isolate scope)